### DISCUSSION TITLE
What's the deal with `useFirestoreEmulator`?

### DISCUSSION THREAD

**Discussion by akauppi** [firebase/firebase-js-sdk#4531] [Q&A] [answered]
I want to set up Firestore emulator, using the `@exp` API.

Docs [Common use cases](https://modularfirebase.web.app/common-use-cases/firestore/#connect-to-the-emulator-suite) show using `initializeFirestore`. But there's a [`useFirestoreEmulator`](https://modularfirebase.web.app/reference/firestore_.usefirestoreemulator) as well. 

Should I use one, either one, or both?

---

*Issue about syntax highlighting [moved here](https://github.com/firebase/firebase-js-sdk/discussions/4534)*

Firebase 0.900.15

**Comment by sam-gc** [comment, marked as answer]
For the emulator functions, this was a conscious decision to have free-floating functions to configure the emulator (so that these don't impact the bundle size when you're not using the emulator). So you'd use (`initializeFirestore` OR `getFirestore`) AND `useFirestoreEmulator`. As with the old version, make sure the emulator function is called before the library is used.

For example:
```js
import { getFirestore, useFirestoreEmulator } from 'firebase/firestore';
const firestore = getFirestore(app);
useFirestoreEmulator(firestore, 'localhost', 8080);
```

(The difference between `initializeFeature` and `getFeature` is that the former allows for more custom configuration of dependencies, whereas the latter pulls in the normal platform-specific defaults).

As for IDE highlighting issue, I'm not sure about that

**Reply by akauppi** [reply]
Your code is how I initially wrote it. Then I noticed the [docs](https://modularfirebase.web.app/common-use-cases/firestore/#connect-to-the-emulator-suite) showing another way.

Thanks for the confirmation. Should I open a different discussion about the IDE highlight - looks like a packaging bug to me.

**Reply by sam-gc** [reply]
Thanks for pointing that out; we need to update the docs. The API was in flux for a little bit there.

Yeah please open up a new discussion specifically for the IDE highlighting :)
